index.js

```diff
@@ -1,37 +1,28 @@
 'use strict';
 
+var constructorRegex = /^\s*class\b/;
 var fnToStr = Function.prototype.toString;
+var objToStr = Object.prototype.toString;
+var hasToStringTag = typeof Symbol === 'function' && typeof Symbol.toStringTag === 'symbol';
 
-var constructorRegex = /^\s*class\b/;
-var isES6ClassFn = function isES6ClassFunction(value) {
-	try {
-		var fnStr = fnToStr.call(value);
-		return constructorRegex.test(fnStr);
-	} catch (e) {
-		return false; // not a function
-	}
-};
+module.exports = function isCallable(value) {
+	if (!value) { return false; } // function can`t be casted to false, so fast-fail
+
+	var valueType = typeof value;
+	if (valueType === 'function') {
+		if (!value.prototype) { return true; } // arrow functions
+	} else if (valueType !== 'object') { return false; } // any else non function-ish types
 
-var tryFunctionObject = function tryFunctionToStr(value) {
 	try {
-		if (isES6ClassFn(value)) { return false; }
-		fnToStr.call(value);
-		return true;
+		var functionString = fnToStr.call(value); // most of non-functions will fail here
```

Comment:
by inlining the try/catch, this actually makes performance much much worse on everything that's not latest v8.

Reply:
Well, this is not a hard one too.
I'll fix it in a moment;

Reply:
what i mean is, the existing library in master afaik is the fastest it can possibly get, taking into account all supported browsers.

Reply:
Sorry if bother, but why not as in my latest aproach?

Because for modern browsers (that has `toStringTag`) it calls `fnToStr.call(value);` twice: inside `isES6ClassFunction` and one more time inside `tryFunctionToStr`.  
Actually this is main reason why it is twice slower than it could be in modern browsers.

Reply:
I agree that there's probably an improvement to be made in terms of calling Function toString, but that wouldn't be this large a rewrite.

Reply:
Well initially i've just written it from scratch, to see if i can squeeze some performance without seing you code, thats why i has so much differences;

About main function changes:
- `lines 21-23`: little economy in strings comparisson, function check made only once instead of original twice compares.
- `lines 25-31`: the only way i see to merge two function calls into one.
- `lines 33-34`: no need to create one more variable if we can reuse existing, so why dont?